urts/src/lib.rs

```diff
@@ -93,35 +179,77 @@ impl EnclaveBuilder {
                 self.debug as c_int,
                 &mut enclave_id,
                 ptr::null_mut(),
-                0,
-                ptr::null_mut(),
+                ex_features,
+                &mut ex_features_p as *mut *const c_void,
             )
-        };
-        match result {
-            sgx_status_t::SGX_SUCCESS => Ok(Enclave::new(enclave_id)),
-            error => Err(Error::SgxStatus(error)),
         }
+        .into_result()
+        .map_err(Error::from)
+        .map(|_| Enclave { id: enclave_id })
     }
 }
 
 impl From<Vec<u8>> for EnclaveBuilder {
-    fn from(bytes: Vec<u8>) -> EnclaveBuilder {
+    fn from(input: Vec<u8>) -> EnclaveBuilder {
         EnclaveBuilder {
-            bytes,
+            bytes: input,
             debug: false,
+            pcl_key: None,
+            kss_config: None,
         }
     }
 }
 
-impl Enclave {
-    fn new(id: sgx_enclave_id_t) -> Enclave {
-        Enclave { id }
+impl From<&[u8]> for EnclaveBuilder {
+    fn from(input: &[u8]) -> EnclaveBuilder {
+        EnclaveBuilder {
+            bytes: input.to_vec(),
+            debug: false,
+            pcl_key: None,
+            kss_config: None,
+        }
+    }
+}
+
+impl<const N: usize> From<&[u8; N]> for EnclaveBuilder {
+    fn from(input: &[u8; N]) -> EnclaveBuilder {
+        EnclaveBuilder {
+            bytes: input.to_vec(),
+            debug: false,
+            pcl_key: None,
+            kss_config: None,
+        }
     }
 }
 
-impl Deref for Enclave {
-    type Target = sgx_enclave_id_t;
-    fn deref(&self) -> &Self::Target {
+impl TryFrom<File> for EnclaveBuilder {
+    type Error = std::io::Error;
+    fn try_from(mut input: File) -> std::io::Result<EnclaveBuilder> {
+        let mut bytes = vec![];
+        input.read_to_end(&mut bytes)?;
+        Ok(EnclaveBuilder {
+            bytes,
+            debug: false,
+            pcl_key: None,
+            kss_config: None,
+        })
+    }
+}
+
+impl Enclave {
+    /// Returns the target info for the enclave.
+    pub fn get_target_info(&self) -> Result<TargetInfo, Error> {
+        let mut target_info = MaybeUninit::uninit();
+        unsafe { sgx_get_target_info(self.id, target_info.as_mut_ptr()) }
+            .into_result()
+            .map_err(Error::from)
+            .map(|_| unsafe { target_info.assume_init() }.into())
+    }
+
+    /// Returns a reference to the enclave ID.
+    /// Returns by reference because enclave ID will not be valid after the
+    /// enclave is dropped.
+    pub fn get_id(&self) -> &sgx_enclave_id_t {
```

Comment:
:wrench: :wrench: rust lang discourages `get_` https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter
This is rust-lang, but most of our conventions usually mimic them.
```suggestion
    pub fn id(&self) -> &sgx_enclave_id_t {
```

Reply:
:+1: